packages/knip/test/plugins/vitest9.test.ts

```diff
@@ -0,0 +1,40 @@
+import { test } from 'bun:test';
+import assert from 'node:assert/strict';
+import { main } from '../../src/index.js';
+import { resolve } from '../../src/util/path.js';
+import baseArguments from '../helpers/baseArguments.js';
+import baseCounters from '../helpers/baseCounters.js';
+
+const cwd = resolve('fixtures/plugins/vitest9');
+
+test('Find dependencies in vitest configuration (projects with inline and external)', async () => {
+  const { issues, counters } = await main({
+    ...baseArguments,
+    cwd,
+  });
+
+  // Should detect unlisted dependencies from both inline and external project configurations
+  // (external configs are loaded dynamically and dependencies attributed to main config)
+  assert(issues.unlisted['vitest.config.ts']['jsdom']);
+  assert(issues.unlisted['vitest.config.ts']['@vitest/coverage-v8']);
+  assert(issues.unlisted['vitest.config.ts']['happy-dom']);
+  assert(issues.unlisted['vitest.config.ts']['@vitest/coverage-istanbul']);
+
+  // Should detect unresolved setup files from both inline and external project configurations
+  // (external configs are loaded dynamically and setup files attributed to main config)
+  assert(issues.unresolved['vitest.config.ts']['./src/unit.setup.ts']);
+  assert(issues.unresolved['vitest.config.ts']['./src/integration.setup.ts']);
+  assert(issues.unresolved['vitest.config.ts']['./src/global.setup.ts']);
+  assert(issues.unresolved['vitest.config.ts']['./e2e-setup.ts']);
+  assert(issues.unresolved['vitest.config.ts']['./spec-setup.ts']);
```

Comment:
If the `toConfig()` suggestion would be applied, the issues should be assigned to the correct workspace.

Ideally, the fixtures are a little bit more minimal: if one file can't be resolved the others won't be resolved either, and if the file is actually there then that would be even faster.